form-data-utils/src/index.ts

```diff
@@ -112,3 +126,79 @@ export function dataFrom(
   return data;
 }
 
+function getSelectValue(field: HTMLSelectElement) {
+  return field.multiple ? getMultipleSelectValue(field) : getSingleSelectValue(field);
+}
+
+function getSingleSelectValue(field: HTMLSelectElement) {
+  // avoid looping if we know nothing is selected
+  if (field.selectedIndex === -1) return null;
+
+  let optionValue: unknown = null;
+
+  for (let opt of field.options) {
+    if (!opt.disabled && opt.selected) {
+      optionValue = getOptionValue(opt);
+    }
+  }
+
+  return optionValue;
+}
+
+function getMultipleSelectValue(field: HTMLSelectElement) {
+  // avoid looping if we know nothing is selected
+  if (field.selectedIndex === -1) return [];
+
+  let optionValues: unknown[] = [];
+
+  for (let opt of field.options) {
+    if (!opt.disabled && opt.selected && opt.value !== '') {
+      optionValues.push(getOptionValue(opt));
+    }
+  }
+
+  return optionValues;
+}
+
+function getOptionValue(opt: HTMLOptionElement) {
+  if (!opt.disabled && opt.selected) {
+    // we normalize empty string to null
+    if (opt.value === '') return null;
+
+    return getValue(opt) || opt.value;
+  }
+}
+
+function getRadioCheckboxValue(el: HTMLInputElement) {
+  if (el.disabled) return;
+
+  // if radio or checkbox were not supplied any value, we assume the user wants a boolean
+  // el.getAttribute('value') returns null when value is not supplied (el.value returns 'on', so we can't use it)
+  const isValueDefined = el.getAttribute('value') !== null || getValue(el);
+
+  if (!el.disabled) {
+    if (!isValueDefined) return getValue(el) || el.checked;
+
+    if (el.checked) {
+      return getValue(el) || el.value;
+    } else {
```

Comment:
```suggestion
    }
```

no need for an `else`